game/missiongenerator/luagenerator.py

```diff
@@ -210,6 +211,49 @@ def generate_plugin_data(self) -> None:
                     iads_element.add_key_value(property, value)
             for role, connections in node.connections.items():
                 iads_element.add_data_array(role, connections)
+        
+        # Add artillery and support units info
+        artillery_object = lua_data.add_item("artilleryGroups")
+        ground_artillery_group_collection = artillery_object.get_or_create_item("groundArtillery")
+        ship_artillery_group_collection = artillery_object.get_or_create_item("shipArtillery")
+
+        # First add all artillery units that are theater objects (mostly ships)
+        for cp in self.game.theater.control_points_for(player=True):
+            for ground_object in cp.ground_objects:
+                for group in ground_object.groups:
+                    # Check if first unit in group is ground-based or ship artillery
+                    unit = group.units[0]
+                    if unit.unit_type is None:
+                        continue
+                    if unit.unit_type.unit_class == UnitClass.ARTILLERY:
+                        ground_artillery_group = ground_artillery_group_collection.add_item()
+                        ground_artillery_group.add_key_value("groupName", group.group_name)
+                    elif unit.unit_type.unit_class in (UnitClass.CRUISER, UnitClass.DESTROYER, UnitClass.FRIGATE):
+                        # TODO: we assume that these ship classes have guns... Which might not be the case.
```

Comment:
Perhaps we could define a JSON-list or something similar to make the arty-capable ships configurable?

Reply:
I agree, but I'm not sure where this would fit in the codebase. It would also be a bit of a research undertaking to do so, plus MBot's scripts already checks the group's ammo and caliber, and only considers rockets and calibers above 70 mm IIRC as possible artillery. I think this would probably rule out the vast majority of cases where ships without guns are added to the artillery groups.

Still, I do think in the long run, having this information somewhere in a JSON would be better. In the meantime, I could probably rename `shipArtillery` to `shipsWithOffensiveArmament` or something like. What do you think?

Reply:
I don't think renaming to `shipsWithOffensiveArmament` will benefit us a lot, but as you say the checks basically happen in the script. I mainly want to avoid registering handlers that're technically unnecessary, since those will waste CPU time while flying. Nevertheless, I think the performance loss is probably going to be minimal, so I guess we can go ahead with this for the time being 👍